lib/logstash/filters/cidr.rb

```diff
@@ -35,11 +35,82 @@ class LogStash::Filters::CIDR < LogStash::Filters::Base
   #     }
   config :network, :validate => :array, :default => []
 
+  # The full path of the external file containing the IP network(s) to check against. Example:
+  # [source,ruby]
+  #     filter {
+  #       %PLUGIN% {
+  #         add_tag => [ "linklocal" ]
+  #         address => [ "%{clientip}" ]
+  #         network_path => "/etc/logstash/networks"
+  #       }
+  #     }
+  # NOTE: it is an error to specify both 'network' and 'network_path'.
+  config :network_path, :validate => :path
+
+  # The separator character used in the encoding of the external file pointed by network_path.
+  config :separator,:validate => :string, :default => "\n"
+
+  # When using a network list from a file, this setting will indicate how frequently
+  # (in seconds) logstash will check the file for updates.
+  config :refresh_interval, :validate => :number, :default => 600
+
   public
-  def register
-    # Nothing
+  def register    
+    rw_lock = java.util.concurrent.locks.ReentrantReadWriteLock.new
+    @read_lock = rw_lock.readLock
```

Comment:
We shouldn't be using just the read lock of this lock object, and definitely not for write operations. Read locks can be held by multiple threads so right now there is no thread safety and all. The translate filter does this correctly by using the read lock for read operations and the write lock for write operations. The same idea should apply here.

That said, since the writing we're doing probably is atomic it's possible that the code doesn't need any locks at all, but it's probably wise to have them anyway since race conditions can easily sneak in and can be very hard to debug.